flocker/acceptance/testtools.py

```diff
@@ -936,9 +936,21 @@ def nodes_available():
         ).write()
 
         def failed_query(failure):
-            Message.new(message_type="acceptance:is_available_error",
-                        reason=unicode(failure),
-                        exception=unicode(failure.__class__)).write()
+            try:
+                # Try to detect _WrapperException.
+                reasons = failure.value.reasons
+            except AttributeError:
+                # Guess it was something else.  Do some simpler logging.
+                system = "acceptance:is_available_error"
+                Message.new(
```

Comment:
:question: `eliot.writeFailure` will include a traceback, if you think that'd be helpful.